includes/Uninstaller.php

```diff
@@ -0,0 +1,174 @@
+
+/**
+ * Uninstaller class.
+ */
+class Uninstaller extends Plugin {
+
+	protected const PREFIX = 'web_stories\_%';
+
+	/**
+	 * Delete network and site option.
+	 *
+	 * @since 1.26.0
+	 */
+	public function remove_options(): void {
+		if ( ! is_multisite() ) {
+			$this->delete_options();
+		}
+		$this->delete_site_options();
+		$site_ids = get_sites(
+			[
+				'fields'                 => 'ids',
+				'number'                 => '',
+				'update_site_cache'      => false,
+				'update_site_meta_cache' => false,
+			]
+		);
+
+		foreach ( $site_ids as $site_id ) {
+			// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.switch_to_blog_switch_to_blog
+			switch_to_blog( $site_id );
+
+			$this->delete_options();
+		}
+
+		restore_current_blog();
+	}
+
+	/**
+	 * Deletes options and transients.
+	 *
+	 * @since 1.26.0
+	 */
+	protected function delete_options(): void {
+		global $wpdb;
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery
+		$options = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT option_name FROM $wpdb->options WHERE option_name LIKE %s",
+				self::PREFIX
+			)
+		);
+
+		if ( ! empty( $options ) ) {
+			array_map( 'delete_option', (array) $options );
+		}
+
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery
+		$transients = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT option_name FROM $wpdb->options WHERE option_name LIKE %s OR option_name LIKE %s",
+				'_transient_' . self::PREFIX,
+				'_transient_timeout_' . self::PREFIX
+			)
+		);
+
+		if ( ! empty( $transients ) ) {
+			array_map( 'delete_option', (array) $transients );
+		}
+	}
+
+	/**
+	 * Deletes options and transients on multisite.
+	 *
+	 * @since 1.26.0
+	 */
+	protected function delete_site_options(): void {
+		global $wpdb;
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery
+		$options = $wpdb->get_results(
+			$wpdb->prepare(
+				"SELECT * FROM $wpdb->sitemeta WHERE meta_key LIKE %s",
+				self::PREFIX
+			)
+		);
+
+		if ( ! empty( $options ) ) {
+			foreach ( (array) $options as $option ) {
+				delete_network_option( $option->site_id, $option->meta_key );
+			}
+		}
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery
+		$transients = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT meta_key FROM $wpdb->sitemeta WHERE meta_key LIKE %s OR meta_key LIKE %s",
+				'_site_transient_' . self::PREFIX,
+				'_site_transient_timeout_' . self::PREFIX
+			)
+		);
+
+		if ( ! empty( $transients ) ) {
+			array_map( 'delete_site_option', (array) $transients );
+		}
+	}
+
+	/**
+	 * Schedule the potential registration of a single service.
+	 *
+	 * This takes into account whether the service registration needs to be delayed or not.
+	 *
+	 * @since 1.26.0
+	 *
+	 * @param string       $id    ID of the service to register.
+	 * @param class-string $class Class of the service to register.
+	 */
+	protected function schedule_potential_service_registration( $id, $class ): void {
+		$this->maybe_register_service( $id, $class );
+	}
```

Comment:
So this is the part where we always register services even they're otherwise `Delayed`?

Would be good to update the doc block here to clarify the difference between this one and the parent method